Makefile

```diff
@@ -93,7 +98,7 @@ test: ## Run tests
 	$(DOCKER_CMD) hack/ci-test.sh
 
 unit:
-	$(DOCKER_CMD) go test ./pkg/... ./cmd/...
+	$(DOCKER_CMD) KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path --bin-dir $(PROJECT_DIR)/bin)" go test $(GOTEST_FLAGS) ./pkg/... ./cmd/...
```

Comment:
I'd probably move the `go test ...` command into its own script so we can be more consistent with the approach we are taking in the control-plane-machine-set-operator repo:
See:
- https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/dad8c9b9bd1e6df0a2f72a3baaeac03f111153bb/hack/test.sh
- https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/dad8c9b9bd1e6df0a2f72a3baaeac03f111153bb/Makefile#L96-L98

and so we can later evolve it from there (e.g. adding the ginkgo test runner (with richer output and tools) and so on). WDYT?

Reply:
Sounds good to me, but i would rather do this in follow up pr, along with another improvements, like ginkgo update

Reply:
👍